crates/ruff_linter/src/rules/flake8_logging/rules/exception_call_outside_handlers.rs

```diff
@@ -0,0 +1,114 @@
+use ruff_diagnostics::{Diagnostic, Edit, Fix, FixAvailability, Violation};
+use ruff_macros::{derive_message_formats, ViolationMetadata};
+use ruff_python_ast::{Expr, ExprAttribute, ExprCall};
+use ruff_python_semantic::analyze::logging;
+use ruff_python_semantic::Modules;
+use ruff_text_size::Ranged;
+
+use crate::checkers::ast::Checker;
+use crate::rules::flake8_logging::rules::helpers::outside_handlers;
+
+/// ## What it does
+/// Checks for `.exception()` logging calls outside of exception handlers.
+///
+/// ## Why is this bad?
+/// [The documentation] states:
+/// > This function should only be called from an exception handler.
+///
+/// Calling `.exception()` outside of an exception handler
+/// attaches `None` as exception information, leading to confusing messages:
+///
+/// ```pycon
+/// >>> logging.exception("example")
+/// ERROR:root:example
+/// NoneType: None
+/// ```
+///
+/// ## Example
+///
+/// ```python
+/// import logging
+///
+/// logging.exception("Foobar")
+/// ```
+///
+/// Use instead:
+///
+/// ```python
+/// import logging
+///
+/// logging.error("Foobar")
+/// ```
+///
+/// ## Fix safety
+/// The fix, if available, will always be marked as unsafe, as it changes runtime behavior.
+///
+/// [The documentation]: https://docs.python.org/3/library/logging.html#logging.exception
+#[derive(ViolationMetadata)]
+pub(crate) struct ExceptionCallOutsideHandlers;
+
+impl Violation for ExceptionCallOutsideHandlers {
+    const FIX_AVAILABILITY: FixAvailability = FixAvailability::Sometimes;
+
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        "`.exception()` call outside exception handlers".to_string()
+    }
+
+    fn fix_title(&self) -> Option<String> {
+        Some("Replace with `.error()`".to_string())
+    }
+}
+
+/// LOG004
+pub(crate) fn exception_call_outside_handlers(checker: &mut Checker, call: &ExprCall) {
+    let semantic = checker.semantic();
+
+    if !semantic.seen_module(Modules::LOGGING) {
+        return;
+    }
```

Comment:
The logger module is the wild west... Some rules test for `semantic.seen_module(Modules::LOGGING)` while others don't. Some test for `is_logger_candidate` while others dont... 

I'm not a 100% sure but I feel like we should either have one or the other because the rule would still get skipped becaues of the module check even if I add e.g. `mylogging.Logger` to logger objects. Wdyt?